### PR TITLE
editoast: add train export command for train schedule v2

### LINKED ISSUE #6735: TS2: Add editoast CLI to export a timetable
## Description and Goal 

Creation of a CLI for exporting trains based on TrainSchedule V2

## Acceptance Criteria

- [ ] add a subcommand: `cargo run -- trains export <TIMETABLE_ID>  <PATH.JSON>`
- [ ] if the given timetable_id does not exist it should fail
- [ ] <PATH.JSON> is required and represents the path to which trains will be stored in json

### PR DESCRIPTION
**Pull request by ghost**
Closes #6735

### CODE REVIEW

File: editoast/src/client/mod.rs [hunk 1]
```diff
@@ -68,6 +69,16 @@ pub struct ImportTrainArgs {
     pub path: PathBuf,
 }
 
+#[derive(Args, Debug, Derivative)]
+#[derivative(Default)]
+#[command(about, long_about = "Export a train given a timetable id")]
```
Comment (flomonster) [inline editoast/src/client/mod.rs]: I missed it in the import PR but all `long_about` and `help` arg can be defined using `///` rust documentation. This way it's both documented and used by clap. We could do this for all commands in another PR for more consistency.

File: editoast/src/main.rs [hunk 2]
```diff
@@ -194,10 +196,56 @@ async fn run() -> Result<(), Box<dyn Error + Send + Sync>> {
...
+    if !missing.is_empty() {
+        let error = CliError::new(
+            1,
+            format!(
+                "❌ Some train schedules were not found, ids: {0:?}",
+                missing
+            ),
+        );
+        return Err(Box::new(error));
+    }
```
Comment (flomonster) [inline editoast/src/main.rs]: This can be transformed into an assert since it's not supposed to happen.

File: editoast/src/main.rs [hunk 3]
```diff
@@ -194,10 +196,56 @@ async fn run() -> Result<(), Box<dyn Error + Send + Sync>> {
...
+
+    let train_schedules: Vec<TrainScheduleBase> =
+        train_schedules.into_iter().map(|ts| ts.into()).collect();
+
+    let train_schedules_json = serde_json::to_string_pretty(&train_schedules)?;
```
Comment (flomonster) [inline editoast/src/main.rs]: We could use [to_writer_pretty](https://docs.rs/serde_json/latest/serde_json/fn.to_writer_pretty.html)

File: editoast/src/schema/v2/trainschedule.rs [hunk 4]
```diff
@@ -35,6 +36,26 @@ pub struct TrainScheduleBase {
     pub options: TrainScheduleOptions,
 }
 
+impl From<TrainSchedule> for TrainScheduleBase {
```
Comment (flomonster) [inline editoast/src/schema/v2/trainschedule.rs]: This shouldn't be done here cause:
- We are trying to keep the From impl in views
- We will soon split editoast in multiple crates. Two of which will be `Schema` and `model`, and the model crate will depend on the schema. 

-------

What can you do:
- Move this in the `modelv2::train_schedule` module
- Remove this impl and use `TrainScheduleResult`. So you can do: `ts.into::<TrainScheduleResult>().train_schedule`

**Review by flomonster — CHANGES_REQUESTED** [review]
Thanks for this PR.
I made a few request changes.
I just realized that the `trains` subcommand is a bit weird in usage, the name `timetable` makes more sense. If you agree with this can you rename the subcommand? (Sorry for the change)

File: editoast/src/main.rs [hunk 5]
```diff
@@ -868,25 +916,41 @@ mod tests {
...
             path: file.path().into(),
             timetable: Some(timetable.id),
         };
-
         let result = trains_import(args, db_pool.clone()).await;
-
         assert!(result.is_ok(), "{:?}", result);
 
+        // Test to export the import
+        let export_file = NamedTempFile::new().unwrap();
```
Comment (Khoyo) [inline editoast/src/main.rs:926]: Make it an expect?

**Review by Khoyo — APPROVED** [review]

File: editoast/src/client/mod.rs (hunk 1 above)
Comment (ghost) [reply editoast/src/client/mod.rs]: ~~Noted, I will open an issue and do this on another PR, I'm going to keep this one this way in order to stay consistent with current code convention, wdyt ?~~

My bad, thought it was something to change to all commands, I'm changing it in this PR.

File: editoast/src/main.rs (hunk 5 above)
Comment (ghost) [reply editoast/src/main.rs:926]: `unwrap()` is used on all other tests, do you think we should change the whole file ?

Comment (Khoyo) [reply editoast/src/main.rs:926]: Yes.

**Review by flomonster — APPROVED** [review]
LGTM
